docs/Compatibility.md

```diff
@@ -4,156 +4,260 @@ The Ruby Datadog Trace library is open source. See the [dd-trace-rb][1] GitHub r
+|       |                            | 2.4       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.3       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.2       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.1       | [EOL](#support-eol) | < 2.0.0             |
+|       |                            | 2.0       | [EOL](#support-eol) | < 0.50.0            |
+|       |                            | 1.9       | [EOL](#support-eol) | < 0.27.0            |
+| JRuby | https://www.jruby.org      | 9.3       | [GA](#support-ga)   | Latest              |
+|       |                            | 9.2.21.0+ | [GA](#support-ga)   | Latest              |
 
 ### Supported web servers
 
-| Type      | Documentation                     | Version      | Support type |
-| --------- | --------------------------------- | ------------ | ------------ |
-| Puma      | http://puma.io/                   | 2.16+ / 3.6+ | Full         |
-| Unicorn   | https://bogomips.org/unicorn/     | 4.8+ / 5.1+  | Full         |
-| Passenger | https://www.phusionpassenger.com/ | 5.0+         | Full         |
+| Type      | Documentation                     | Version      | Support type      |
+|-----------|-----------------------------------|--------------|-------------------|
+| Puma      | http://puma.io/                   | 2.16+ / 3.6+ | [GA](#support-ga) |
+| Unicorn   | https://bogomips.org/unicorn/     | 4.8+ / 5.1+  | [GA](#support-ga) |
+| Passenger | https://www.phusionpassenger.com/ | 5.0+         | [GA](#support-ga) |
 
 ### Supported tracing frameworks
 
-| Type        | Documentation                                   | Version               | Gem version support |
-| ----------- | ----------------------------------------------- | --------------------- | ------------------- |
-| OpenTracing | https://github.com/opentracing/opentracing-ruby | 0.4.1+                | < 2.0.0            |
+| Type        | Documentation                                   | Version | Support type        | Gem version support |
+|-------------|-------------------------------------------------|---------|---------------------|---------------------|
+| OpenTracing | https://github.com/opentracing/opentracing-ruby | 0.4.1+  | [EOL](#support-eol) | < 2.0.0             |
 
-*Full* support indicates all tracer features are available.
+### Supported operating systems
 
-*Deprecated* indicates support will transition to *Maintenance* in a future release.
+| OS           | Support type                        | Package version |
+|--------------|-------------------------------------|-----------------|
+| Linux x86-64 | [GA](#support-ga)                   | Latest          |
+| Linux ARM64  | [GA](#support-ga)                   | Latest          |
+| macOS        | Dev environments only               | Latest          |
```

Comment:
> adding one column listing the list of platforms for each OS

Maybe this can be two columns: cpu arch and libc

Which makes me think the macOS "dev env only" in "support type" is a bit weird, maybe "environment" should be its own column so that this can read "GA".

Also "support type" and "package version" feel redundant, it seems like one implies the other (and conversely).

```
| OS      | CPU             | libc      | environment | Support type                        |
|---------|-----------------|-----------|-------------|-------------------------------------|
| Linux   | x86_64, aarch64 | gnu, musl | prod, dev   | [GA](#support-ga)                   |
| macOS   | x86_64, arm64   | darwin    | dev         | [GA](#support-ga)                   |
```

Thoughts?

Reply:
> Both x86_64-darwin and arm64-darwin are supported.

The current 1.0 compatibility doc states 

```Use of ddtrace on macOS is supported for development, but not for production deployments.```

I just moved that statement to fit a tabular format. It is my understanding that while ddtrace may run, we don't officially support macOS in production. 

If that is not the case, I can update the doc. @delner is there more to the story about why we don't support macOS in production?

Reply:
I'll create a ticket so we can track enhancing this table in the near future.